### PR TITLE
Move common types to types.h

### PR DESCRIPTION
**Pull request by gordonwang0**
Moves shared types to interfaces.h

### CODE REVIEW

File: libraries/standard/common/include/types.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
```
Comment (sarenameas) [inline libraries/standard/common/include/types.h:1]: Do we really want a common header that is shared by multiple libraries like this? It looks like an abstraction layer and adds to what needs to be consumed if taking MQTT by itself out of the CSDK.

Comment (gordonwang0) [reply libraries/standard/common/include/types.h:1]: I thought we decided not to duplicate the types?  It does make it harder to separate the libraries though.  I'll close this.

Comment (sarenameas) [reply libraries/standard/common/include/types.h:1]: I think my concern is more of a discussion point on how we want libraries consumed. I think this opens up some unanswered questions on how libraries are to be consumed. I am not sure if everyone shares what my interpretation. We should talk about it in a team meeting.

Comment (abhidixi11) [reply libraries/standard/common/include/types.h:1]: We must move fixed buffer to MQTT specific header where the context would reside.  But I'm fine with sharing the transport interface as long as the signature is the same. Do you see any problem in sharing the transport interface ?

**Pull request closed without merging**